app/controllers/courses_controller.rb

```diff
@@ -12,21 +18,43 @@ def show
   end
 
   def update
+    if !current_user.admin?
```

Comment:
Good call. I also think its important to be explicit. Better to check twice than to have someone edit the file in the future and take out the admin check. In fact, I was thinking, we want to specify a bunch of routes at a higher level as being admin only.